readme.md

```diff
@@ -164,6 +164,14 @@ In the repository you will find instructions and build tools to compile and run
 
 If you are considering porting the device client SDK for C to a new platform, check out the [porting guide](https://github.com/Azure/azure-c-shared-utility/blob/master/devdoc/porting_guide.md) document.
 
+## Deprecation Notes
```

Comment:
Nit1 - we need a TOC entry at top of the readme.md pointing to this for consistency.
Nit2 - consider opening a separate GitHub issue to track eventually removing this from the main readme.  Agree it makes sense now but 2 years off, especially given how little the package based was used, we can potentially remove this to keep the opening readme as straightforward as it can be.  Analogous to #2417 master=>main rename.

Reply:
Added the TOC entry.
Added a PBI (16775354) in the C SDK backlog for tracking the removal of MBED OS entry from Deprecation Notes later in 2024.